### ISSUE TITLE
Error when generating passcode

### ISSUE THREAD

**Issue by ndahlquist** [twilio-labs/plugin-rtc#16]
### Issue Summary
I am attempting to generate a passcode as described here: https://github.com/twilio/twilio-video-app-android#deploy-twilio-access-token-server

### Steps to Reproduce
1. Run `twilio rtc:apps:video:deploy --authentication passcode -l debug`

### Exception/Log
```
nicolas.dahlquist@mp-nicolasd-ml ~/w/twilio-video-app-react> twilio rtc:apps:video:deploy --authentication passcode -l debug
[DEBUG] Config File: /Users/nicolas.dahlquist/.twilio-cli/config.json
[DEBUG] Using profile: testaccount
[DEBUG] -- BEGIN Twilio API Request --
[DEBUG] GET https://serverless.twilio.com/v1/Services
[DEBUG] User-Agent: @twilio/cli-core/4.5.0 (twilio-node/3.41.1, node.js v13.11.0, darwin 18.7.0 x64, rtc:apps:video:deploy)
[DEBUG] -- END Twilio API Request --
[DEBUG] response.statusCode: 200
[DEBUG] response.headers: {"date":"Thu, 19 Mar 2020 04:41:33 GMT","content-type":"application/json; charset=utf-8; charset=utf-8","content-length":"1067","connection":"close","twilio-concurrent-requests":"1","twilio-request-id":"RQef3f3b82ff6d4f649c2934dca6d2d093","twilio-request-duration":"0.020","access-control-allow-origin":"*","access-control-allow-headers":"Accept, Authorization, Content-Type, If-Match, If-Modified-Since, If-None-Match, If-Unmodified-Since","access-control-allow-methods":"GET, POST, DELETE, OPTIONS","access-control-expose-headers":"ETag","access-control-allow-credentials":"true","x-powered-by":"AT-5000","x-shenanigans":"none","x-home-region":"us1","x-api-domain":"serverless.twilio.com","strict-transport-security":"max-age=31536000"}
[DEBUG] -- BEGIN Twilio API Request --
[DEBUG] GET https://serverless.twilio.com/v1/Services/ZSd53e3a289ad47670c6ac2159386227ff/Environments
[DEBUG] User-Agent: @twilio/cli-core/4.5.0 (twilio-node/3.41.1, node.js v13.11.0, darwin 18.7.0 x64, rtc:apps:video:deploy)
[DEBUG] -- END Twilio API Request --
[DEBUG] response.statusCode: 200
[DEBUG] response.headers: {"date":"Thu, 19 Mar 2020 04:41:33 GMT","content-type":"application/json; charset=utf-8; charset=utf-8","content-length":"380","connection":"close","twilio-concurrent-requests":"1","twilio-request-id":"RQaba4f24e6cdc48c18fe38f89f33e7470","twilio-request-duration":"0.025","access-control-allow-origin":"*","access-control-allow-headers":"Accept, Authorization, Content-Type, If-Match, If-Modified-Since, If-None-Match, If-Unmodified-Since","access-control-allow-methods":"GET, POST, DELETE, OPTIONS","access-control-expose-headers":"ETag","access-control-allow-credentials":"true","x-powered-by":"AT-5000","x-shenanigans":"none","x-home-region":"us1","x-api-domain":"serverless.twilio.com","strict-transport-security":"max-age=31536000"}
 » twilio-cli encountered an unexpected error. To report this issue, execute the command with the "-l debug" flag, then copy the output to a new issue here: https://github.com/twilio/twilio-cli/issues
[DEBUG] Cannot read property 'sid' of undefined
[DEBUG] TypeError: Cannot read property 'sid' of undefined
    at DeployCommand.getAppInfo (/Users/nicolas.dahlquist/.twilio-cli/node_modules/@twilio-labs/plugin-rtc/src/helpers.js:74:64)
    at processTicksAndRejections (internal/process/task_queues.js:97:5)
    at async DeployCommand.run (/Users/nicolas.dahlquist/.twilio-cli/node_modules/@twilio-labs/plugin-rtc/src/commands/rtc/apps/video/deploy.js:18:21)
    at async DeployCommand._run (/Users/nicolas.dahlquist/.twilio-cli/node_modules/@oclif/command/lib/command.js:44:20)
    at async Config.runCommand (/usr/local/Cellar/twilio/1.11.0/libexec/node_modules/@oclif/config/lib/config.js:160:9)
    at async Main.run (/usr/local/Cellar/twilio/1.11.0/libexec/node_modules/@oclif/command/lib/main.js:21:9)
    at async Main._run (/usr/local/Cellar/twilio/1.11.0/libexec/node_modules/@oclif/command/lib/command.js:44:20)
```

### Technical details:
* twilio-cli version: twilio-cli/1.11.0
* node version: node-v13.11.0
* plugin-rtc: v0.1.2
* darwin-x64

This issue was originally reported at https://github.com/twilio/twilio-cli/issues/160.

**Comment by aaalaniz** [comment]
Hey @ndahlquist 

Do you happen to have the following environment variables set on your machine?

```
TWILIO_ACCOUNT_SID
TWILIO_AUTH_TOKEN
TWILIO_API_KEY
TWILIO_API_SECRET
```

Can you try unsetting the values and using what is generated at `~/.twilio-cli/config.json`?

Thanks!

**Comment by ndahlquist** [comment]
@aaalaniz I don't seem to have any of those variables set: `env | grep -i TWILIO` returns nothing.

```
cat ~/.twilio-cli/config.json 
{"email":{},"prompts":{},"projects":[{"id":"testaccount","accountSid":"[REDACTED]"}],"activeProject":null}
```

**Comment by aaalaniz** [comment]
Hey @ndahlquist 

Maybe I have this wrong, but can you try setting the following environment variables?

```
TWILIO_ACCOUNT_SID
TWILIO_AUTH_TOKEN
```

These should match the credentials used when you called `twilio login`.

I recently, encountered this same error and fixed it by clearing all my TWILIO_* environment variable, deleting the API keys created by the Twilio CLI in the Twilio Console, and re-configuring the Twilio CLI. 

Once we get this resolved, we will update the README to reflect how to remediate this issue.

**Comment by ndahlquist** [comment]
Okay, I've set those two environment variables, but I'm still seeing the same error:

```
nicolas.dahlquist@mp-nicolasd-ml ~/w/twilio-video-app-react> env | grep -i TWILIO
PWD=/Users/nicolas.dahlquist/workspace/twilio-video-app-react
TWILIO_ACCOUNT_SID=[REDACTED]
TWILIO_AUTH_TOKEN=[REDACTED]
nicolas.dahlquist@mp-nicolasd-ml ~/w/twilio-video-app-react> twilio rtc:apps:video:deploy --authentication passcode -l debug
[DEBUG] Config File: /Users/nicolas.dahlquist/.twilio-cli/config.json
[DEBUG] Using profile: ${TWILIO_ACCOUNT_SID}/${TWILIO_AUTH_TOKEN}
[DEBUG] -- BEGIN Twilio API Request --
[DEBUG] GET https://serverless.twilio.com/v1/Services
[DEBUG] User-Agent: @twilio/cli-core/4.5.0 (twilio-node/3.41.1, node.js v13.11.0, darwin 18.7.0 x64, rtc:apps:video:deploy)
[DEBUG] -- END Twilio API Request --
[DEBUG] response.statusCode: 200
[DEBUG] response.headers: {"date":"Thu, 19 Mar 2020 20:01:42 GMT","content-type":"application/json; charset=utf-8; charset=utf-8","content-length":"1067","connection":"close","twilio-concurrent-requests":"1","twilio-request-id":"RQ0419af174b4f47229a6b99acc9c1d190","twilio-request-duration":"0.028","access-control-allow-origin":"*","access-control-allow-headers":"Accept, Authorization, Content-Type, If-Match, If-Modified-Since, If-None-Match, If-Unmodified-Since","access-control-allow-methods":"GET, POST, DELETE, OPTIONS","access-control-expose-headers":"ETag","access-control-allow-credentials":"true","x-powered-by":"AT-5000","x-shenanigans":"none","x-home-region":"us1","x-api-domain":"serverless.twilio.com","strict-transport-security":"max-age=31536000"}
[DEBUG] -- BEGIN Twilio API Request --
[DEBUG] GET https://serverless.twilio.com/v1/Services/ZSd53e3a289ad47670c6ac2159386227ff/Environments
[DEBUG] User-Agent: @twilio/cli-core/4.5.0 (twilio-node/3.41.1, node.js v13.11.0, darwin 18.7.0 x64, rtc:apps:video:deploy)
[DEBUG] -- END Twilio API Request --
[DEBUG] response.statusCode: 200
[DEBUG] response.headers: {"date":"Thu, 19 Mar 2020 20:01:43 GMT","content-type":"application/json; charset=utf-8; charset=utf-8","content-length":"380","connection":"close","twilio-concurrent-requests":"1","twilio-request-id":"RQ4e3cb45c6b524322a7f7b4ee96d4afc3","twilio-request-duration":"0.029","access-control-allow-origin":"*","access-control-allow-headers":"Accept, Authorization, Content-Type, If-Match, If-Modified-Since, If-None-Match, If-Unmodified-Since","access-control-allow-methods":"GET, POST, DELETE, OPTIONS","access-control-expose-headers":"ETag","access-control-allow-credentials":"true","x-powered-by":"AT-5000","x-shenanigans":"none","x-home-region":"us1","x-api-domain":"serverless.twilio.com","strict-transport-security":"max-age=31536000"}
 » twilio-cli encountered an unexpected error. To report this issue, execute the command with the "-l debug" flag, then copy the output to a new issue here: https://github.com/twilio/twilio-cli/issues
[DEBUG] Cannot read property 'sid' of undefined
[DEBUG] TypeError: Cannot read property 'sid' of undefined
    at DeployCommand.getAppInfo (/Users/nicolas.dahlquist/.twilio-cli/node_modules/@twilio-labs/plugin-rtc/src/helpers.js:74:64)
    at processTicksAndRejections (internal/process/task_queues.js:97:5)
    at async DeployCommand.run (/Users/nicolas.dahlquist/.twilio-cli/node_modules/@twilio-labs/plugin-rtc/src/commands/rtc/apps/video/deploy.js:18:21)
    at async DeployCommand._run (/Users/nicolas.dahlquist/.twilio-cli/node_modules/@oclif/command/lib/command.js:44:20)
    at async Config.runCommand (/usr/local/Cellar/twilio/1.11.0/libexec/node_modules/@oclif/config/lib/config.js:160:9)
    at async Main.run (/usr/local/Cellar/twilio/1.11.0/libexec/node_modules/@oclif/command/lib/main.js:21:9)
    at async Main._run (/usr/local/Cellar/twilio/1.11.0/libexec/node_modules/@oclif/command/lib/command.js:44:20)
```

One of my coworkers was able to generate a passcode though, so I'm unblocked now.

**Comment by timmydoza** [comment]
Thanks for the info @ndahlquist! We will be looking into the cause of the issue you reported, but in the meantime, I have a tip to help you get unstuck. 

You can use twilio's serverless CLI plugin to take a closer look at things. First, install it:
`twilio plugins:install @twilio-labs/plugin-serverless`

Then, list any deployed services you might have:
`twilio api:serverless:v1:services:list`

You might see one called `video-app`. If so, you can remove it with this:
`twilio api:serverless:v1:services:remove --sid <SID of video-app service>`

Now, you should be able to deploy the app again with `twilio rtc:apps:video:deploy`
I hope it helps!

**Comment by mosesoak** [comment]
@timmydoza Thank you so much for your last post. This totally was the problem. 

The cryptic error about `TypeError: Cannot read property 'sid' of undefined` would never have led me to understand that an existing serverless instance was already connected to my account and needed to be manually deleted. I wonder if there would be some way in fixing this issue to detect this case and provide an error message indicating that a serverless app already exists and need to be deleted. (Using the `--override` flag on deploy did not work around this problem.)

To be fair, the tutorial lists running `twilio rtc:apps:video:delete` which seems to work the same as your method. I could have sworn I tried doing that a few times though so, not sure what caused the issue for me, or if there are cases where only your method would work.

**Comment by aaalaniz** [comment]
Hey @mosesoak 

Glad @timmydoza's guide helps! We have a ticket in our backlog to update the `delete` command to remove the service even when no environments are present. It appears this issue is a result of a Twilio Serverless Client bug. We should be able to work around it though.

Thanks for you input and we'll let you know when we update the delete command!

**Comment by mosesoak** [comment]
Fantastic!!

**Comment by jonoirwinrsa** [comment]
Just a heads up - I was running into the same `[DEBUG] Cannot read property 'sid' of undefined` issue as above. Your fix worked though but I was quite lost there for a while. It would be nice to have this in the readme if the real fix is a little way off?

**Comment by timmydoza** [comment]
The `twilio:apps:video:delete` command has been updated to remove the app, even if there are no environments. This should allow users to fix their deployments when they see the `Cannot read property 'sid' of undefined` error.

To upgrade the cli plugin, run `twilio plugins:update`.

Also, we have identified the root cause of the issue, and the Serverless team here at Twilio is working on a fix. It should be out in a few days.
